nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -405,21 +405,25 @@ public boolean updateNotRunningWorkflowInstance(WorkflowInstance instance) {
   }
 
   @Transactional
-  public boolean wakeUpWorkflowExternally(int workflowInstanceId) {
-    String sql = "update nflow_workflow set next_activation = (case when executor_id is null then "
+  public boolean wakeUpWorkflowExternally(int workflowInstanceId, String[] expectedStates) {
+    StringBuilder sql = new StringBuilder("update nflow_workflow set next_activation = (case when executor_id is null then "
         + "least(current_timestamp, coalesce(next_activation, current_timestamp)) else next_activation end), "
         + "external_next_activation = current_timestamp where " + executorInfo.getExecutorGroupCondition()
-        + " and id = ? and next_activation is not null";
-    return jdbc.update(sql, workflowInstanceId) == 1;
+        + " and id = ? and next_activation is not null");
+    return addExpectedStatesToQueryAndUpdate(sql, workflowInstanceId, expectedStates);
   }
 
-  public boolean wakeupWorkflowInstanceIfNotExecuting(long id, String[] expectedStates) {
+  public boolean wakeupWorkflowInstanceIfNotExecuting(long workflowInstanceId, String[] expectedStates) {
```

Comment:
Forget this, WorkflowService.wakeupWorkflowInstance() is useful for users of nFlow.